dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -0,0 +1,45 @@
+from io import open, StringIO, BytesIO
+
+from . import data_utils
+
+class FileOrBufferHandler():
+    """
+    FileOrBufferHandler class to read a filepath or buffer in and always return a readable buffer
+    """
+    def __init__(self, io_obj, open_method='r', encoding=None, seek_offset=None, seek_whence=0):
+        """
+        Context manager class used for inputing a file or buffer and returning a structure
+        that is always a buffer.
+
+        :param io_obj: path to the file being loaded or buffer
+        :type io_obj: string, StringIO, or BytesIO 
+        :param open_method: value describes the mode the file is opened in
+        :type open_method: string
+        :param seek_offset: offset from start of the stream
+        :type seek_offset: int
+        :return: TextIOBase or BufferedIOBase class/subclass
+        """
+        self._io_obj = io_obj
+        self.open_method = open_method
+        self.seek_offset = seek_offset
+        self.seek_whence = seek_whence
+        self._encoding = encoding
+
+    def __enter__(self):
+        if isinstance(self._io_obj, str):
+            self._io_obj = open(self._io_obj, self.open_method, encoding=self._encoding)
+
+        elif not isinstance(self._io_obj, (StringIO, BytesIO)):
+            # Raise AttributeError if attribute value not found.
+            raise AttributeError(f'Type {type(self._io_obj)} is invalid.')
```

Comment:
We should specifically say what is wrong as in why, e.g. it was not a string or a stream buffer